lib/agent0/agent0/accounts_config.py

```diff
@@ -0,0 +1,190 @@
+"""Defines the accounts configuration from env vars."""
+from __future__ import annotations
+
+import json
+import logging
+import os
+import sys
+from dataclasses import dataclass
+
+import numpy as np
+from dotenv import load_dotenv
+from fixedpointmath import FixedPoint
+
+from .base.config import AgentConfig, Budget
+from .base.make_key import make_private_key
+
+
+@dataclass
+class AccountKeyConfig:
+    """The account config dataclass
+
+    Attributes
+    ----------
+    ARTIFACTS_URL: str
+        The url of the artifacts server from which we get addresses.
+    RPC_URL: URI | str
+        The url to the ethereum node
+    ABI_DIR: str
+        The path to the abi directory
+    """
+
+    # default values for local contracts
+    # Matching environment variables to search for
+    # pylint: disable=invalid-name
+    USER_KEY: str | None
+    AGENT_KEYS: list[str]
+    AGENT_ETH_BUDGETS: list[int]
+    AGENT_BASE_BUDGETS: list[int]
+
+    def to_env_str(self):
```

Comment:
I think you can `return json.dumps(self.__dict__)` since it's a `dataclass`. At that point, maybe it makes sense to just make it a property?
```
@property
def env_str(self) -> str:
    return json.dumps(self.__dict__)
```

Reply:
I guess you need to handle None. You could add `None` as a case in the [elfpy ExtendedJSONEncoder](https://github.com/delvtech/elf-simulations/blob/main/lib/elfpy/elfpy/utils/json.py#L11). We can package up utility stuff like this together out of elfpy later.

Reply:
This function is being used to write an .env file, which I don't think json is compatible with